saleor/payment/gateways/adyen/webhooks.py

```diff
@@ -582,6 +595,8 @@ def handle_webhook(request: WSGIRequest, gateway_config: "GatewayConfig"):
         return HttpResponseBadRequest("Invalid or missing hmac signature.")
     if not validate_auth_user(request.headers, gateway_config):
         return HttpResponseBadRequest("Invalid or missing basic auth.")
+    if not validate_merchant_account(notification, gateway_config):
```

Comment:
If we got a notification for different merchant than ours, we should log it, skip it and return `[accepted]`. If we return `HttpResponseBadRequest`, Adyen will try to send the request one more time and again and again.
Also, I would move this `if ` before validation of the hmac and auth. If the merchant account is different, there is a big possibility that hmac and user would be different.